File: javascripts/Scenes/PowerDemocracy.js
function PowerDemocracy(session){
	this.canRepeat = true;
	this.playerList = [];  //what players are already in the medium when i trigger?
	this.session = session;
	//a player has to be not busy to be your friend right now.
	this.trigger = function(playerList){
		this.playerList = playerList;


		return (this.session.democracyStrength > 0);
	}

	this.renderContent = function(div){
		div.append("<br><img src = 'images/sceneIcons/wv_icon.png'>"+this.content());
	}


	this.content = function(){
		var ret = "";
		var rand = Math.seededRandom();
		if(rand < .25){
			ret += getRandomElementFromArray(democracyTasks);
			this.session.democraticArmy.power += 10;
			this.session.democraticArmy.mobility += 10;
			this.session.democraticArmy.currentHP += 10;
		}else if(rand < .5){
			ret += getRandomElementFromArray(democracyTasks);
			this.session.democraticArmy.power += 20;
			this.session.democraticArmy.mobility += 20;
			this.session.democraticArmy.currentHP += 20;
		}else if(rand < .75){
			ret += getRandomElementFromArray(democracySuperTasks);
   		this.session.democraticArmy.power += 50;
			this.session.democraticArmy.mobility += 50;
			this.session.democraticArmy.currentHP += 50;
		}else{
			//do nothing.
			ret += getRandomElementFromArray(mayorDistractionTasks);
		}
		return ret;
	}
}

File: scripts/Scenes/PowerDemocracy.dart
part of SBURBSim;


class PowerDemocracy extends Scene{
	bool canRepeat = true;
	List<dynamic> playerList = [];  //what players are already in the medium when i trigger?



	PowerDemocracy(Session session): super(session);

	@override
	bool trigger(playerList){
		this.playerList = playerList;


		return (this.session.democracyStrength > 0);
	}

	@override
	void renderContent(Element div){
    appendHtml(div,"<br><img src = 'images/sceneIcons/wv_icon.png'>"+this.content());
	}
	dynamic content(){
		String ret = "";
		double r = rand.nextDouble();
		if(r < .25){
			ret += rand.pickFrom(democracyTasks);
			this.session.democraticArmy.addStat("power", 10);
			this.session.democraticArmy.addStat("mobility", 10);
			this.session.democraticArmy.addStat("currentHP", 10);
		}else if(r < .5){
			ret += rand.pickFrom(democracyTasks);
      this.session.democraticArmy.addStat("power", 20);
      this.session.democraticArmy.addStat("mobility", 20);
      this.session.democraticArmy.addStat("currentHP", 20);
		}else if(r < .75){
			ret += rand.pickFrom(democracySuperTasks);
      this.session.democraticArmy.addStat("power", 50);
      this.session.democraticArmy.addStat("mobility", 50);
      this.session.democraticArmy.addStat("currentHP", 50);
		}else{
			//do nothing.
			ret += rand.pickFrom(mayorDistractionTasks);
		}
		return ret;
	}

}
